### PR TITLE
[TEP074] Tombstone ResourceResult field with the removal of PipelineResources

### PR DESCRIPTION
**Pull request by JeromeJu**
# Changes
This commit tombstones the `resourcesResult` field of TaskRun along with the removal of `pipelineResources`.
It aims to cleanup `pkg/pod/status` as the prerequisites swapping to v1 as storage version.

It is tombstoned due to following reasons:
- `ResourceResult`, according its description, was 'Results from Resources built during the TaskRun. currently includes the digest of build container images', now since image resources are removed, it was left meaningless.
- `ResourceResult` highly relies on PipelineResourcesResult type, where previously PipelineResourceResult is being written to it, now that PipelineResources is being removed, there is no usage for resourcesResult
- `ResourceResult` has not been moved to v1, when we swap the storage versions, there will not be a suitable place to write to `PipelineResourceResult` any more

The migration of PipelineResourceResult is broken down into:
- Stop populating resourceName in git-init image https://github.com/tektoncd/pipeline/pull/6310
- removing `PipelineResourceResultType` of `PipelineResourceResult` Struct (https://github.com/tektoncd/pipeline/pull/6198)
- tombstoning of deprecated field `resourcesResult` (https://github.com/tektoncd/pipeline/pull/6301)

/kind misc
part of https://github.com/tektoncd/pipeline/issues/6197

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [n/a] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by afrittoli** [comment]
This makes sense since we don't have `PipelineResources` anymore.
I'm slightly concerned that we did not explicitly deprecate this field, so the removal might break users. 
However, since we deprecated `PipelineResources` a long time ago though, and this field is only populated by `PipelineResources`, it feels like we gave quite some notice already. 

To be on the safe side, we could mark it as deprecated in the code, and remove it in the following release?
@tektoncd/core-maintainers opinions?

**Comment by abayer** [comment]
Based on past cases where we’ve removed fields in the status and the pain that’s caused with downstream consumers of the status structs, I’d be conservative with the removal.

**Comment by JeromeJu** [comment]
> Based on past cases where we’ve removed fields in the status and the pain that’s caused with downstream consumers of the status structs, I’d be conservative with the removal.

Thanks @abayer , I agree with the cautions that we need to take when dealing with api fields. But when I tried removing `ResourcesResult`, it didn't feel like much functionalities related apart from the `PipelineResourceResult` writing to it. Would it be reasonable that we tombstone this field in the current release whilst remove the usage in `pkg/pod/status` and then we remove the field in the following release? Or I would love to have advice/plans if there are other precautions to be taken 🙏

**Comment by JeromeJu** [comment]
> This makes sense since we don't have `PipelineResources` anymore. I'm slightly concerned that we did not explicitly deprecate this field, so the removal might break users. However, since we deprecated `PipelineResources` a long time ago though, and this field is only populated by `PipelineResources`, it feels like we gave quite some notice already.
> 
> To be on the safe side, we could mark it as deprecated in the code, and remove it in the following release? @tektoncd/core-maintainers opinions?

  Appreciate the help @afrittoli , I also added the reasons why this was attempted to be removed: (also in the PR comment)
  - `ResourceResult`, according its description, was 'Results from Resources built during the TaskRun. currently includes the digest of build container images', now since image resources are removed, it was left meaningless.
- `ResourceResult` highly relies on PipelineResourcesResult type, where previously PipelineResourceResult is being written to it, now that PipelineResources is being removed, there is no usage for resourcesResult
- `ResourceResult` has not been moved to v1, when we swap the storage versions, there will not be a suitable place to write to `PipelineResourceResult` any more

File: pkg/apis/pipeline/v1beta1/taskrun_types.go
Comment (wlynch) [inline pkg/apis/pipeline/v1beta1/taskrun_types.go:273]: I'd retain this field for backwards compatibility. We never marked it as deprecated, and similar to the TaskRun status we should avoid removing this until we're sure most clients will no longer need this for older objects / server versions.

**Review by wlynch — CHANGES_REQUESTED** [review]

File: pkg/apis/pipeline/v1beta1/taskrun_types.go
Comment (JeromeJu) [reply pkg/apis/pipeline/v1beta1/taskrun_types.go]: Thanks @wlynch for the guidance from the users' perspective.
totally understand the concerns for backwards compatibility concerns for older versions. It was debating for this field as it is related with PipelineResources that has long deprecated and just been removed.

For moving forward, would changing from **removal** to **tomb-stoning** the `taskrun.status.resourcesresult` field make sense in this PR as the followup of migrating off the usage of the `PipelineResourceResult` struct etc as the plan for `PipelineResourceResult` migration?

File: pkg/apis/pipeline/v1beta1/task_types.go
Comment (wlynch) [inline pkg/apis/pipeline/v1beta1/task_types.go:31]:
```suggestion
	// reserved: 2
	// was PipelineResourceResultType
	
	// InternalTektonResultType default internal tekton result value
	InternalTektonResultType = 3
```

I'd revert this too and just add a comment that 2 was PipelineResourceResultType but was removed.

This is used for deserialization (if you're familiar with proto field numbers, it serves a similar purpose). This could confuse clients that have pods mid-flight during upgrade.

File: pkg/apis/pipeline/v1beta1/taskrun_types.go
Comment (JeromeJu) [reply pkg/apis/pipeline/v1beta1/taskrun_types.go]: And shall we announce deprecation/ tomestoning this field since the resourcesResult is not used any where in the codebase, and it won't interfere with the usage:)

**Review by wlynch — CHANGES_REQUESTED** [review]

File: pkg/apis/pipeline/v1beta1/taskrun_types.go
Comment (wlynch) [reply pkg/apis/pipeline/v1beta1/taskrun_types.go]: SGTM! We should definitely tombstone + mark as deprecated :)

File: pkg/apis/pipeline/v1beta1/task_types.go
Comment (JeromeJu) [reply pkg/apis/pipeline/v1beta1/task_types.go:31]: ahhh makes sense, thanks for pointing that out

**Comment by pritidesai** [comment]
I could not recall how `results` from resources looked like in a taskRun for a task with pipeline resources. Any older tests that you might have seen such results?

**Comment by JeromeJu** [comment]
> I could not recall how `results` from resources looked like in a taskRun for a task with pipeline resources. Any older tests that you might have seen such results?

I might not fully understood but is this referring to `PipelineResourceResultType` of `ResourcesResult` for taskruns?
If so I think they are at https://github.com/tektoncd/pipeline/blob/ce2e9b95ca03bf4b7d2c278ae758b7b038918ca8/pkg/pod/status_test.go#L674-L677 though some of them are not specified as `PipelineResourceResultType` :/

File: docs/deprecations.md
Comment (lbernick) [inline docs/deprecations.md:29]: I would add this under "deprecated PipelineResources related features" instead of here

File: pkg/pod/status.go
Comment (lbernick) [inline pkg/pod/status.go:256]: can you add a docstring for this function? I'm confused why pipelineresourceresults are still used here-- can we update the code that uses them?

**Comment by JeromeJu** [comment]
/hold 
for the following PRs that should get in before the current one
[Stop populating resourceName in git-init image](https://github.com/tektoncd/pipeline/pull/6310)
[Migrate git-init image off PipelineResourceResult](https://github.com/tektoncd/pipeline/pull/6335)

**Comment by pritidesai** [comment]
This PR is waiting for https://github.com/tektoncd/pipeline/pull/6335 which was not part of 0.46 for some reason.

Moving this to the next milestone 0.47 along with PR #6335

File: docs/deprecations.md
Comment (JeromeJu) [reply docs/deprecations.md]: Thanks Lee. Moved the the section below.

File: pkg/pod/status.go
Comment (lbernick) [inline pkg/pod/status.go:304]: since `pipelineresourceresults` aren't returned, the variable can be removed

File: pkg/pod/status.go
Comment (lbernick) [reply pkg/pod/status.go:263]: A good docstring should explain all of the inputs and all of the outputs, for example:

```
filterResultsAndResources filters the input results based on the results declared in the task spec.
It returns a slice of any of the input results that are defined in the task spec, converted to TaskRunResults,
and a slice of any of the input results that don't represent internal values (i.e. those that should not be displayed in the TaskRun status.
```

**Comment by JeromeJu** [comment]
> filterResultsAndResources filters the input results based on the results declared in the task spec.
It returns a slice of any of the input results that are defined in the task spec, converted to TaskRunResults,
and a slice of any of the input results that don't represent internal values (i.e. those that should not be displayed in the TaskRun status.

Thanks for the suggestions for the doc string, I was trying to rename the `PipelineResourceResult` first since we are removing resources related structs in this PR. And also to rename this function 🤔 

Should have put a hold on this PR to wait for #6434 to merge.

/hold

**Comment by JeromeJu** [comment]
/unhold
https://github.com/tektoncd/pipeline/pull/6434 is merged.

File: pkg/pod/status_test.go
Comment (lbernick) [inline pkg/pod/status_test.go:650]: I wonder if it would make sense to keep this test case, but just have it not populate anything in resourcesresult?

**Review by lbernick — APPROVED** [review]

File: pkg/pod/status_test.go
Comment (JeromeJu) [reply pkg/pod/status_test.go:650]: Thanks Lee. restored.

**Comment by dibyom** [comment]
@JeromeJu I don't think this needs an action-required release note since there is nothing actionable for the Tekton user. This is an internal cleanup part of the pipeline resources removal, and pipeline resources have already been removed.

**Review by vdemeester — APPROVED** [review]

**Pull request merged**
